BlackBoxPredictor OSS part 5: glow transforms

salexspb opened the pull request:
Summary:
Overal context: open-source BlackBoxPredictor as the entry
point for inference in Caffe2 (thread safe abstraction for Caffe2
inference). This should be used in ThroughputBenchmark for the purpose
of framework comparison
This specific diff:
There should be no harm in moving transformation code to
OSS. On the advantages side we will be able to compare production
Caffe2 setup with PyTorch in the most fair way via
ThroughputBenchmark. This approach avoid any complicated
transformation regirstries. Building those proper would be significant
engineering effort as well as production risk. In the past we had SEVs
related to transforms being turned off due to various refactors. Given
that we don't plan to build any other significant investments into
transformation logic except existing ones (like TVM and Glow), and
those also relate to open-source technologies, I came up to the
conclusion of moving to OSS the whole thing.

Differential Revision: D16367134

bwasti commented on caffe2/utils/string_utils.h:
```diff
@@ -10,7 +10,8 @@
 
-CAFFE2_API std::vector<std::string> split(char separator, const std::string& string);
+CAFFE2_API std::vector<std::string>
+split(char separator, const std::string& string, bool ignore_empty = false);
```
is this change necessary? I don't see a use of the third variable in the code

salexspb replied:
That was for the glow transform, it originally relied on this but it was gone via rebase. I think they want to recover it back.

salexspb committed:
BlackBoxPredictor OSS part 5: glow transforms

Summary:
Overal context: open-source BlackBoxPredictor as the entry
point for inference in Caffe2 (thread safe abstraction for Caffe2
inference). This should be used in ThroughputBenchmark for the purpose
of framework comparison
This specific diff:
There should be no harm in moving transformation code to
OSS. On the advantages side we will be able to compare production
Caffe2 setup with PyTorch in the most fair way via
ThroughputBenchmark. This approach avoid any complicated
transformation regirstries. Building those proper would be significant
engineering effort as well as production risk. In the past we had SEVs
related to transforms being turned off due to various refactors. Given
that we don't plan to build any other significant investments into
transformation logic except existing ones (like TVM and Glow), and
those also relate to open-source technologies, I came up to the
conclusion of moving to OSS the whole thing.

Reviewed By: bertmaher

Differential Revision: D16367134

fbshipit-source-id: 74e9732476714846876b0a0ef79748f57fcda798
